tests/server.rs

```diff
@@ -553,6 +553,42 @@ fn no_proto_nonempty_parse_eof_returns_error() {
     core.run(fut).unwrap_err();
 }
 
+// Test that the server does not exit with "too many open files" IO errors with
+// a lot of TCP connections.
+#[test]
+fn no_ports_exhausted_errors() {
+    let server = serve();
+    let addr = server.addr();
+
+    let mut core = Core::new().unwrap();
+    let handle = core.handle();
+
+    // Send 100k requests (TCP connections).
+    let nr_requests = 100_000;
+    let concurrency = 10_000;
+
+    let mut parallel = Vec::new();
+    for _i in 0..concurrency {
+        let requests_til_done = loop_fn(0, |counter| {
+            // Just establish the TCP connection, do nothing otherwise.
+            let socket = tokio_core::net::TcpStream::connect(&addr, &handle);
+
+            socket.then(move |_| -> Result<_, std::io::Error> {
```

Comment:
It seems that this will drop the `TcpStream` right upon connect, and so there won't actually be that many concurrent connections. I think.

Reply:
Good point! Not sure what we can do to keep the connection open for a bit before dropping it? Add a future that just sleeps 1 second?